master/internal/configpolicy/task_config_policy_intg_test.go

```diff
@@ -0,0 +1,84 @@
+package configpolicy
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+)
+
+func TestPriorityAllowed(t *testing.T) {
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	// When no constraints are present, any priority is allowed.
+	ok, err := PriorityAllowed(1, model.NTSCType, 0, true)
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	wkspLimit := 50
+	w := addWorkspacePriorityLimit(t, pgDB, wkspLimit)
+
+	// Priority is outside workspace limit.
+	ok, err = PriorityAllowed(w.ID, model.NTSCType, wkspLimit-1, true)
+	require.NoError(t, err)
+	require.False(t, ok)
+
+	globalLimit := 42
+	addGlobalPriorityLimit(t, pgDB, globalLimit)
+
+	// Priority is within global limit.
```

Comment:
```suggestion
	// Priority is outside workspace limit but within global limit.
```

Reply:
Because of the order of precedence, it doesn't matter what the workspace limit is, right? The global limit is applied first (workspace limit is only used if global limit does not exist).

Reply:
Fair point! I was more suggesting this for clarity of the test case, but this is definitely a matter of preference